src/Nethermind/Nethermind.Network/PeerPool.cs

```diff
@@ -101,6 +105,17 @@ private Peer CreateNew(PublicKeyAsKey key, (Node Node, ConcurrentDictionary<Publ
         private Peer CreateNew(PublicKeyAsKey key, (NetworkNode Node, ConcurrentDictionary<PublicKeyAsKey, Peer> Statics) arg)
         {
             Node node = new(arg.Node);
+
+            string enodeString = node.ToString(Node.Format.ENode);
+
+            Enode enode = new Enode(enodeString);
+
+            // Check if this node is trusted
+            if (_trustedNodesManager.IsTrusted(enode))
+            {
+                node.IsTrusted = true;
+            }
+
```

Comment:
```suggestion
            Node node = new(arg.Node) { IsTrusted = _trustedNodesManager.IsTrusted(arg.Node.NodeId) };
```

Reply:
You need to expose `Enode` property from `NetworkNode` (only `get`):

```suggestion
            Node node = new(arg.Node) { IsTrusted = _trustedNodesManager.IsTrusted(arg.Node.Enode) };
```